management/utils.py

```diff
@@ -179,6 +179,35 @@ def wait_for_service(port, public, env, timeout):
 				return False
 		time.sleep(min(timeout/4, 1))
 
+def get_ssh_port():
+	port_value = get_ssh_config_value("port")
+	
+	if port_value:
+		return int(port_value)
+
+	return None
+
+def get_ssh_config_value(parameter_name):
+	# Returns ssh configuration value for the provided parameter
+	try:
+		output = shell('check_output', ['sshd', '-T'])
+	except FileNotFoundError:
+		# sshd is not installed. That's ok.
+		return None
+	except subprocess.CalledProcessError:
+		# error while calling shell command
+		return None
+
+	returnNext = False
+	for e in output.split():
```

Comment:
The structure of your primary processing loop, compares every item in the output to the "parameter_name" that is being sought. This also means you compare the parameter_name to values, not just keys. Also, the logic assumes there's only 1 value belonging to a key. That is true most, but not all, of the time.

The output of ssh -T is as follows:
```
key1 val1 [val2 ... valN]
key2 val1 [val2 ... valN]
...
```

Most keys can only have 1 value, but some can have multiple values, so I would propose 2 changes to your loop logic here.

First, you should loop over each line of the output, treating each as a record instead of looping over every item in the output. For each line, the first item is the key and all items after that are values. Then you would only compare the parameter_name to the first item (the key), if they match return the list of values (see the next point). If the parameter_name does not match, move on to the next line.

Second, the function (to be complete) really should return a list of values, even though 1 value is far more common. Alternatively, your function could just return a single value, but then you will need to come up with a way of dealing with the few cases where there are more than 1 value.

Here's a possible outline to handle both single and multiple value returns:
```
def get_ssh_config_value(parameter_name):
    r = get_ssh_config_values(parameter_name)
    if r is None:
        return None
    else:
        return r[0]

def get_ssh_config_values(parameter_name):
    # Do the work here
    # return None if key not found, else return list containing 1 or more values
```
This suggestion leaves it up to the caller to decide if they want a scalar or list to be returned.

Reply:
To be honest I just reused some code that was already there 😄
But you are right of course, by generalizing the  function name, the code should also have been generalised. It'll work now for the usage within MiaB, but not for all cases.
When I have some more time I'll look into it.